scripts/tests_analyzer/compare_coderabbit_decisions.py

```
@@ -0,0 +1,689 @@
+def compare_pr(
+    repo: str,
+    pr: dict,
+    token: str | None = None,
+) -> ComparisonResult:
+    """Compare CodeRabbit vs Analyzer decision for a single PR."""
+    pr_number = pr["number"]
+    pr_title = pr["title"]
+    pr_url = pr["html_url"]
+    pr_author = pr["user"]["login"]
+
+    logger.info(f"Processing PR #{pr_number}: {pr_title[:50]}...")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)